xmovie/core.py

```diff
@@ -358,12 +370,66 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
             )
 
+
+
+    def save_frames_parallel(self, odir, progress=False, 
+                             parallel_compute_kwargs=dict()):
+        '''
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        progress : Bool
+            Whether or not to show a progress bar.
+        parallel_compute_kwargs : dict
+            Keyword arguments to pass t dask's `compute()` function.
+
+        '''
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+        
+        da = da.chunk({self.framedim : 1}) # DataArray needs to have length-1 chunks in framedim
+        chunk_dims = [ dim for dim in da.dims if dim != framedim ] # Create dimensions for each chunk
+               
+
+        total_time = da[framedim]
+        def create_render_save_single_frame2(xr_array, framedim):
```

Comment:
I find the name of the function a bit hard to interpret. Maybe something like `_save_single_frame_parallel` is clearer?

Reply:
I agree! That name was temporary and I just forgot to change it. Thanks for catching that. 
Done!